### ISSUE TITLE
[java] DuplicateImports reported for the same import... and import static...

### ISSUE THREAD

**Issue by pzygielo** [pmd/pmd#2546]
**Affects PMD Version:** [6.24.0](https://pmd.github.io/latest/pmd_release_notes.html#24-may-2020---6240)

**Rule:** [DuplicateImports](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#duplicateimports)

**Description:**

Violation of DuplicateImports reported for (star) imports of types and statics from other class.

**Code Sample demonstrating the issue:**
https://github.com/pzrep/pmd-DuplicateImports

```
import charles.X.*;
import static charles.X.*;
```
is reported as violation, although they import different things, and one does not replace another.

**Expected outcome:**

No violation of *DuplicateImports* reported.

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

I think so.

<sup>*   Is PMD missing to report a violation, where there should be one? -> false-negative
No.</sup>

**Running PMD through:** CLI in sample, found with maven.

**Comment by pzygielo** [comment]
Thank you for the fix and new release.

**Comment by oowekyala** [comment]
This popped up again while improving the rule in #3718. I think the issue is not real. In the example, the first import is redundant, as the static import already imports the member types. The JLS even has [a paragraph](https://docs.oracle.com/javase/specs/jls/se17/html/jls-7.html#jls-7.5.2) about this situation:

> If a compilation unit contains both a type-import-on-demand declaration and a static-import-on-demand declaration (§7.5.4) that name the same class or interface, the effect is as if the static member classes and interfaces of that class or interface (§8.5, §9.5) are imported only once.

**Comment by pzygielo** [comment]
> This popped up again while improving the rule in #3718. I think the issue is not real. In the example, the first import is redundant, as the static import already imports the member types. The JLS even has [a paragraph](https://docs.oracle.com/javase/specs/jls/se17/html/jls-7.html#jls-7.5.2) about this situation:
> ...
> UnnecessaryImport should indeed report the first import in the example

Unclear.
If I remove `import charles.X.*;` I get
```
src/eric/Magneto.java:12: error: cannot find symbol
    ChessBoard board = professor.setBoard(); // Inner type imported with star
    ^
  symbol:   class ChessBoard
  location: class Magneto
```

If I remove `import static charles.X.*;` I get
```
src/eric/Magneto.java:10: error: cannot find symbol
      = createProfessor();                   // class method imported with star
        ^
  symbol:   method createProfessor()
  location: class Magneto
```

so for me it looks like both imports are needed (at least for `javac`).

**Comment by oowekyala** [comment]
Hmm I get it. Your `ChessBoard` is a non-static class right? In that case you're right, it cannot be imported by the static import...

**Comment by pzygielo** [comment]
> Hmm I get it. Your `ChessBoard` is a non-static class right? In that case you're right, it cannot be imported by the static import...

That's it - indeed, inner, non-static.
